sei-db/sc/memiavl/db.go

```diff
@@ -295,17 +294,23 @@ func (db *DB) ApplyUpgrades(upgrades []*proto.TreeNameUpgrade) error {
 
 // ApplyChangeSets wraps MultiTree.ApplyChangeSets, it also appends the changesets in the pending log,
 // which will be persisted to the rlog in next Commit call.
-func (db *DB) ApplyChangeSets(changeSets []*proto.NamedChangeSet) error {
+func (db *DB) ApplyChangeSets(changeSets []*proto.NamedChangeSet) (_err error) {
 	if len(changeSets) == 0 {
 		return nil
 	}
 
-	db.mtx.Lock()
-	defer db.mtx.Unlock()
 	startTime := time.Now()
 	defer func() {
-		metrics.SeiDBMetrics.ApplyChangesetLatency.Record(context.Background(), time.Since(startTime).Milliseconds())
+		otelMetrics.ApplyChangesetLatency.Record(
```

Comment:
We probably want this before the `if len(changeSets) == 0 {` _if_ we care about knowing the number of calls to apply changeset? the `zeroLen` could be an attribute to differentiate between zero length and non zero.

If you think we don't need to know the number of calls then please feel free to ignore this comment

Reply:
Don't think we need number of calls